Tests/NIOPosixTests/SALChannelTests.swift

```diff
@@ -322,6 +322,48 @@ final class SALChannelTest: XCTestCase, SALTest {
                 }
         }.salWait())
     }
+    
+    func testBindFailureClosesChannel() {
```

Comment:
This test isn't specific enough: it continues to pass even with the new `close0` line removed. It looks like this is because the failed `bind` promise is caught by the `ClientBootstrap` in the `connect` flow and triggers a `close0`.

The same code appears to be present in the server bootstrap. Can we try to nail down where the `bind` issue came from in your original reproducer? It doesn't currently look like `bind0` needs to close where it does.

Reply:
I see, I will be taking a closer look at this over the weekend!